src/client/scripts/generators/entry_generators.js

```diff
@@ -1,10 +1,16 @@
-export function generateDeckEntry(deck) {
+import {User} from "../structures/user.js";
+import {Deck} from "../structures/deck.js"
+
+const activeUser = User.getActiveUser();
+
+export function generateDeckEntry(deck, view) {
     /**
      * Generates a Deck Entry (Icon to represent a deck at a glance)
      * @ToDo
      * Change the text on the button
      * Add click functionality to the button
-     * @argument {Deck} deck - A deck object as defined in structures/deck.js
+     * @param {Deck} deck - A deck object as defined in structures/deck.js
+     * @param {boolean} view - Decides what the top button does. True for study, false for add.
```

Comment:
I'm not sure if we should add this extra view parameter in. If you want to determine if you should place a study button, or the add button, cant you get this information via the deck object itself ? This looks like what you did in another section of the code below.
` studyingButton.addEventListener("click", async () => {
        if (activeUser.metadata[deck.id].beingStudied) {
            studyingButton.value = "Pending";
            await activeUser.toggleStudy(deck);
            studyingButton.value = "Not studying";
        } else {
            studyingButton.value = "Pending";
            await activeUser.toggleStudy(deck);
            studyingButton.value = "Studying";
        }
    })`

Reason I bring this up, is because just having a single argument is simpler, and we wont have to go back and change instances of generateDeckEntry throughout the code.

Reply:
The beingStudied part of the study info is meant to indicate whether that deck is being actively studied and should be considered when you go to the study tab/sort that way. The idea behind the view parameter is that the difference between a study entry and an add entry depends on where the entry is placed rather than the deck itself. If you're looking at another user's profile, it's going to be add, since you're looking at other people's decks to find new ones. If you're on the decks page, you're looking at decks you've created/already added, and thus want the button to be study. I could change it to be dependent on whether or not the deck is already in the user's study data if you think that'd be better.

Reply:
Here is how I was thinking of it 

Case 1 : Deck parameter does not belong to current active user
- this means the passed deck id is not within the active user's decks
- append an add button to the entry
- also append a study button to the entry 

Case 2: Deck parameter belongs to current active user
- this means the passed deck id is within the active user's decks
- append a study button to the entry 

The study button will always have the same behavior
- If the user clicks it, and they do not 'own' the deck, then it is added to their list of owned decks, and toggled as 'isStudying', and they are subsequently taken to the Study page
- If the user clicks it and they do own the deck but are not studying it, toggle it as studying and take them to the study page
- is the user is already studying the deck, take them to the study page

Tell me what you think / any questions.

Reply:
That makes sense to me. I'll make teh changes to get rid of the view parameter.